libs/remix-ui/debugger-ui/src/lib/debugger-ui.tsx

```diff
@@ -181,7 +181,7 @@ export const DebuggerUI = (props: DebuggerUIProps) => {
       return
     }
 
-    const web3 = state.opt.debugWithLocalNode ? await debuggerModule.web3() : await debuggerModule.getDebugWeb3()
+    const web3 = state.opt.debugWithLocalNode ? await debuggerModule.web3() : (optWeb3 || await debuggerModule.getDebugWeb3())
```

Comment:
I think that if `optWeb3` is present, it should have predominance over `await debuggerModule.web3()`

Reply:
updated